Fix boxcox to return same result for same input in one batch

Yuhta opened the pull request:
Summary:
The SIMD path is using SLEEF version of `pow` which is slightly different from `std::pow`.  The fix is to use the same vectorized code (with partial load and store) for the trailing data as well to ensure consistency between results.

Rollback Plan:

Differential Revision: D82265247

swolchok reviewed (changes requested):
Review automatically exported from Phabricator review in Meta.

Yuhta committed:
[koski] Fix boxcox to return same result for same input in one batch (#162772)

Summary:

The SIMD path is using SLEEF version of `pow` which is slightly different from `std::pow`.  The fix is to use the same vectorized code (with partial load and store) for the trailing data as well to ensure consistency between results.

Test Plan:
```
buck test koski/functions_contrib/df4ai/tests:box_cox_test -- SameInputSameOutput
buck2 test fbcode//caffe2/caffe2/python/operator_test:batch_box_cox_test
```

Rollback Plan:

Differential Revision: D82265247
